### PR TITLE
[14.0][ADD] shopinvader_easy_binding

### PR DESCRIPTION
**Pull request by ivantodorovich**
This module lets you manage Shopinvader bindings to a company's main backend in a simpler way.
It makes it a lot easier for the users to manage bindings, if they are only using 1 Shopinvader backend per company.

To configure, go to **Shopinvader > Settings** and choose your **Company Shopinvader Backend**.

After that, the user can Publish / Unpublish products directly from the product form view.

image

### CODE REVIEW

File: shopinvader_company_backend/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2021 Camptocamp (https://www.camptocamp.com).
+# @author Iván Todorovich <ivan.todorovich@gmail.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Shopinvader Company Backend",
+    "summary": "Easily manage Shopinvader bindings to your company backend.",
```
Comment (simahawk) [inline shopinvader_company_backend/__manifest__.py]:
```suggestion
    "summary": "Easily manage Shopinvader bindings for your company backend.",
```

**Review by simahawk — COMMENTED** [review]
LG, I'm not convinced the name is the right one tho. 
I'll wait for others' feedback.

**Comment by ivantodorovich** [comment]
Yeah, I'm not convinced about the name neither :/

**Comment by PierrickBrun** [comment]
Maybe naming it `shopinvader_single_company_backend` is clearer ?

**Review by kevinkhao — APPROVED** [review]
functional test
Works as expected
for names, maybe the idea of "fast binding" expresses the functionality's idea

**Comment by ivantodorovich** [comment]
Thanks everyone !

I'm not thrilled by any name so for now I think I'll keep it as is, unless there's consensus 😅

**Comment by simahawk** [comment]
I vote for `shopinvader_product_easy_bind` which would serve as a base for any improvement on this topic.

File: shopinvader_company_backend/models/product_product.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    def _shopinvader_bind(self, backend):
+        BindingWizard = self.env["shopinvader.variant.binding.wizard"]
+        wizard = BindingWizard.create(
+            {
+                "backend_id": backend.id,
+                "product_ids": [(6, 0, self.ids)],
+                "run_immediately": True,
+            }
+        )
+        wizard.bind_products()
```
Comment (sebastienbeau) [inline shopinvader_company_backend/models/product_product.py]: Note you can also can directly the method "bind_selected_products" of backend, as the wizard will just call this method

```suggestion
        backend.bind_selected_products(self, run_immediately=True)
```

**Review by sebastienbeau — APPROVED** [review]
LGTM code review
minor proposal of change

regarding the naming shopinvader_product_easy_bind seem to be a good idea

**Comment by ivantodorovich** [comment]
What about `shopinvader_easy_binding` ?

Reason is: currently this module only implementing the "main company backend" feature on products, but it could be extended to any kind of binding in fact, even ones added by other modules.

**Comment by ivantodorovich** [comment]
Done.

CI is failing because of the changes done on this PR:
- https://github.com/OCA/brand/pull/87

They will be fixed after this is finished and merged:
- https://github.com/shopinvader/odoo-shopinvader/pull/1088

**Pull request merged**